asmcli/commands/experimental/vm.sh

```diff
@@ -0,0 +1,438 @@
+
+FLAGS:
+  -v|--verbose                        Print commands before and after execution.
+     --dry_run                        Print commands, but don't execute them.
+     --only_validate                  Run validation, but don't install.
+  -h|--help                           Show this message and exit.
+EOF
+}
+
+validate_vm_args() {
+  local PROJECT_ID; PROJECT_ID="$(context_get-option "PROJECT_ID")"
+  local CLUSTER_NAME; CLUSTER_NAME="$(context_get-option "CLUSTER_NAME")"
+  local CLUSTER_LOCATION; CLUSTER_LOCATION="$(context_get-option "CLUSTER_LOCATION")"
+  local SERVICE_ACCOUNT; SERVICE_ACCOUNT="$(context_get-option "SERVICE_ACCOUNT")"
+  local KEY_FILE; KEY_FILE="$(context_get-option "KEY_FILE")"
+  local DRY_RUN; DRY_RUN="$(context_get-option "DRY_RUN")"
+  local ONLY_VALIDATE; ONLY_VALIDATE="$(context_get-option "ONLY_VALIDATE")"
+  local VERBOSE; VERBOSE="$(context_get-option "VERBOSE")"
+
+  local MISSING_ARGS; MISSING_ARGS=0
+  while read -r REQUIRED_ARG; do
+    if [[ -z "${!REQUIRED_ARG}" ]]; then
+      MISSING_ARGS=1
+      warn "Missing value for ${REQUIRED_ARG}"
+    fi
+    readonly "${REQUIRED_ARG}"
+  done <<EOF
+CLUSTER_LOCATION
+CLUSTER_NAME
+PROJECT_ID
+EOF
+
+  if [[ "${MISSING_ARGS}" -ne 0 ]]; then
+    fatal_with_usage "Missing one or more required options."
+  fi
+
+  while read -r FLAG; do
+    if [[ "${!FLAG}" -ne 0 && "${!FLAG}" -ne 1 ]]; then
+      fatal "${FLAG} must be 0 (off) or 1 (on) if set via environment variables."
+    fi
+    readonly "${FLAG}"
+  done <<EOF
+DRY_RUN
+ONLY_VALIDATE
+VERBOSE
+EOF
+
+  if [[ -n "$SERVICE_ACCOUNT" && -z "$KEY_FILE" || -z "$SERVICE_ACCOUNT" && -n "$KEY_FILE" ]]; then
+    fatal "Service account and key file must be used together."
+  fi
+
+  # since we cd to a tmp directory, we need the absolute path for the key file
+  # and yaml file
+  if [[ -f "${KEY_FILE}" ]]; then
+    KEY_FILE="$(apath -f "${KEY_FILE}")"
+    readonly KEY_FILE
+  elif [[ -n "${KEY_FILE}" ]]; then
+    fatal "Couldn't find key file ${KEY_FILE}."
+  fi
+}
+
+validate_vm_cli_dependencies() {
+  local NOTFOUND; NOTFOUND="";
+  local EXITCODE; EXITCODE=0;
+
+  info "Checking installation tool dependencies..."
+  while read -r dependency; do
+    EXITCODE=0
+    hash "${dependency}" 2>/dev/null || EXITCODE=$?
+    if [[ "${EXITCODE}" -ne 0 ]]; then
+      NOTFOUND="${dependency},${NOTFOUND}"
+    fi
+  done <<EOF
+$AGCLOUD
+curl
+jq
+tr
+awk
+grep
+$AKUBECTL
+EOF
+
+  if [[ -n "${NOTFOUND}" ]]; then
+    NOTFOUND="$(strip_last_char "${NOTFOUND}")"
+    for dep in $(echo "${NOTFOUND}" | tr ' ' '\n'); do
+      warn "Dependency not found: ${dep}"
+    done
+    fatal "One or more dependencies were not found. Please install them and retry."
+  fi
+
+  # shellcheck disable=SC2064
+  trap "$(shopt -p nocasematch)" RETURN
+  shopt -s nocasematch
+  if [[ "$(uname -m)" != "x86_64" ]]; then
+    fatal "Installation is only supported on x86_64."
+  fi
+}
+
```

Comment:
You should just be able to use the existing `validate_cli_dependencies`, all of the dependencies there are either optional or included in the same packages as these.

Reply:
Done